### ISSUE TITLE
Completely rewriting Manager with Bootstrap concept

### ISSUE THREAD

**Issue by pmfx** [modxcms/evolution#690]
I believe we need to update Manager look and feel. That's why I was experimenting with rewriting completely Manager with Bootstrap3 over the weekend. A lot of work but it is possible.

To have it all in Bootstrap, huge amount of code will have to be rewritten. I mean HTML in 95%. Mostly files in /manager/actions/ I wonder: will so many changes make it not possible to merge with current develop branch? If so, how to do this? Or **is it possible with normal commits to develop branch**?

For now I have moved (merged) menu.php directly to 1.php to have fully working navbar with dropdowns. I think tree iframe and main iframe can stay as as they are. **Everything is working fine** (updated JS to avoid menu iframe) - example: **saving resource updates tree as expected**. Maybe Tree can be moved but I dont know if its only updated by AJAX or by reloading too.

**ManagerManager** rules like "mm_hideFields" "mm_renameField" **may not work as html will change**. We will have to **update MM**. In new version .form-group divs (wrapping label, input, optional help text) will have unique ID so it will be very easy to target them with MM. **So MM update should be relatively easy**.

# What is working right now:

- Login page
- Dashboard
- Navbar
- Edit resource

All edit resource form elements have bee rewritten with Bootstrap markup. TVs too.

NOTE: **full responsive** Manager (available for mobile devices) **is not the goal**. Goal is new look and feel, easy to mantain Manager code, easier form manipulation etc. Some main frame content uses and will use Bootstrap responsive grid and will nicely collapse on lower resolutions.

Additionally, I see future Evolution with **one solid, consistent theme only** - no need for code supporting 3rd party themes. It will simplify things I think. Revolution is a good example here - only one theme and it is more than ok.

I wonder what core dev team members think about. Let me know @yama @Jako @dmi3yy @Deesen and others.

# Screenshots:

- Dashboard http://prnt.sc/bw2j1v
- Edit resource 1 http://prnt.sc/bw2jgg
- Edit resource 2 http://prntscr.com/bw2tcr
- Edit resource 3 http://prntscr.com/bw2sxg
- Edit resource 4 help popover https://snag.gy/Wg73ZN.jpg

# Demo:
http://pm-fx.com/sandbox/evobootstrap/manager/
user: githubviewer
password: gview345

I have moved this issue from #672 as it is different matter.

**Comment by pmfx** [comment]
Probably resource edit page two column layout would be replaced with one column. Two columns could activated and orginized via new ManagerManager rule. Taken from here https://github.com/modxcms/evolution/issues/672#issuecomment-234520570

**Comment by Fr3ddy7** [comment]
idea a good, but too much padding in the elements. Now everything is great, everything in the default skin is much more logical and takes up less space.

let the same TV they became huge

**Comment by fourroses666** [comment]
@Fr3ddy7  That kind of stuff can be changed pritty simple. Its just basic bootstrap now.

**Comment by fourroses666** [comment]
@pmfx 

Was looking for a different subnav and bumped on this example;
http://www.bootply.com/WjsCLRE3d6

Only on mobile its showing below the normal nav so might be useless.

**Comment by pmfx** [comment]
Resource overview rewritten:
http://prnt.sc/bw70wt
http://prnt.sc/bw713v
http://prnt.sc/bw718j

@Fr3ddy7 I don't think there is too much padding/margins. Remember that we have bigger screens today then in 2005. I am OK with new sizes - all is easy to read and navigate, Remember that a lot of fields will be hidden by MM in most cases. As for TVs I think it is good idea to separate somehow every TV like a mini-fieldset. TVs can be very original especially with custom TVs so they need to be grouped. Anyway, as @fourroses666 said - all elements are easy to style with Bootstrap, so we can play with sizes, margin, paddings as we want at any time.

@fourroses666 why would we need different navbar? Bootstrap one is really nice if you ask me and fits great in Manager needs.

**Comment by fourroses666** [comment]
Correct, its the same to see dropdown or the subnav row. So nvm that one . Really starting to look very nice!

**Comment by nick0** [comment]
I really like the padding / margins you have chosen @pmfx, I think it looks excellent and very modern. Really great.

The template variable section looks so nice. If this is adopted I would hope that we could hide the [*tv-name*] labels (and maybe also the tv-type) with ManagerManager as that would just confuse non technical editors and will save some screen real estate.

I love the question mark for help content, that looks so slick and much better than the current bubble in 1.0.15. There is lot of help text shown in the screencap "bw2sxg". Could the question mark be used to show/hide this text too as an option instead to streamline the info on screen?

Love that overview screen "bw70wt". Much easier to read at a glance than currently.

I also really like the reinterpretation of the top menu. As long as it is easy to access all the various sub sections, that will work great.

Brilliant work @pmfx, looking super :)

**Comment by pmfx** [comment]
@nick0 Thank you! 

> If this is adopted I would hope that we could hide the [tv-name] labels (and maybe also the tv-type) with ManagerManager as that would just confuse non technical editors and will save some screen real estate.

I think it is done already. Actual TV name (tag) is only seen for users that have template editing permissions. But I'm not 100% sure. Think I seen code for this while editing mutate_content.dynamic.php

> There is lot of help text shown in the screencap "bw2sxg". Could the question mark be used to show/hide this text too as an option instead to streamline the info on screen?

I will go back and see what can be done.
Cheers.

**Comment by fourroses666** [comment]
pmfx, just something that fits better into the Bootstrap;
if you would like to change the tree icons you can use these font-awesome icons;

_fa-long-arrow-down / fa-long-arrow-up / fa-refresh / fa-sort-amount-desc / fa-file-image-o / fa-files-o / fa-th / fa-trash_  

or when thats kinda impossible to change the images could also be replaced offcourse.

**Comment by pmfx** [comment]
@fourroses666 I am fully aware of FontAwesome (using it in most of my projects!) but this time I dont see a reason to replace [Glyphicons bundled with Bootstrap.](http://getbootstrap.com/components/#glyphicons) What would be benefits here? Glyphicons contains probably all icons that Manager needs. And it is a lot smaller when it comes to file size.

**Comment by pmfx** [comment]
And yes - I am going to replace image icons in TreeButtons and maybe in Tree too :)

**Comment by pmfx** [comment]
TreeButtons new style with Glyphicons:
Normal: https://snag.gy/r7PKbJ.jpg
Hover: https://snag.gy/C4IsMO.jpg + on hover sidebar have different background color and border on right side.

**Comment by fourroses666** [comment]
Wasn't aware of these default BS icons ⭕ Looks good :)

**Comment by pmfx** [comment]
Tree sort popup: https://snag.gy/y9z36T.jpg

@fourroses666 I've aligned action buttons left because it plays better with other elements. It has something to do with how our eyes read content. Feels easier to do so when I compare two options.

**Comment by fourroses666** [comment]
Getting better and better. I think this is a really good default/base theme. 
_Still would like to see some less high resource page. Having these fields in General all together and no seperate "blocks" like template and link options._

Keep it up. 👍

**Comment by pmfx** [comment]
@fourroses666 here you are: http://prnt.sc/bxgsb8 :)

**Comment by fourroses666** [comment]
Like that! 👊

**Comment by nick0** [comment]
That login looks really nice exactly like that

**Comment by nick0** [comment]
Looking awesome @pmfx!

**Comment by pmfx** [comment]
@fourroses666 I think stacked fields works fine here - no need to make one on left second on right. Your example ooks good. I've merged it with mine and this is result.

![screenshot_1](https://cloud.githubusercontent.com/assets/10888055/17136920/62e8dc5a-5338-11e6-93d8-ec03abee28de.png)

**Comment by fourroses666** [comment]
Much better without the white box.

_I made the footer-message fixed to the bottom and remove the br and not sure whats the latest Logo of MODX but._

**Comment by pmfx** [comment]
@fourroses666 aaaaand done! Background image fits well. Updated some colours. Thanks for spotting double classes. I'll leave tree buttons as they are not distracting user like green ones would.

**Comment by fourroses666** [comment]
👍

**Comment by pmfx** [comment]
I have done some updates to edit page - mostly to help messages and settings tab and thats all for now. I will wait for core dev team thoughts about all this and my first post in this thread.

![screenshot_1](https://cloud.githubusercontent.com/assets/10888055/17153806/9b3d2200-537e-11e6-96d5-6694f1f40722.png)
![screenshot_2](https://cloud.githubusercontent.com/assets/10888055/17153807/9b3e364a-537e-11e6-9343-4d52dcc9b752.png)
![screenshot_3](https://cloud.githubusercontent.com/assets/10888055/17153809/9b40796e-537e-11e6-8cb7-6c34ba748097.png)

**Comment by fourroses666** [comment]
Its looking nice and clean :) 

_These ? are pritty heavy but I can't find a better one beside the i or use other colors which looks good._ Really nice to have the text next to the fields like that.

**Comment by nick0** [comment]
Awesome work! looks excellent.

**Comment by Fr3ddy7** [comment]
Now its cool )

**Comment by Fr3ddy7** [comment]
alternatively I propose to change the icons in the tree :+1:

**Comment by fourroses666** [comment]
As far as I know the core dev team is dmi3yy, deesen and yama.
I think they need to deside together if we're going to use it. 

As new downloadable theme or default theme. 
The old theme(s) must work too I guess, or needs some (small) css changes.
Or there should be a new version 1.2 with breaking changes?

It still needs some improvements but as long as its not getting a "go" this will not go anywhere I'm afraid.

**Comment by Deesen** [comment]
@pmfx: Maybe best approach would be to build the new template the same way as template MODxPress. If you download it and look to its directory-structure you see you can overwrite almost every manager-file with a custom one. You have directories "actions", "frames" etc. for example -> https://github.com/WorkForFood/MODxPress/tree/master/manager/media/style/MODxPress

This way we have the old relyable template, and can work on the new one until its finished without breaking anything.

The only problem (and reason why I proposed a template-engine) is, that if the main files in /manager/actions/ get refactored or updated like by Yama yesterday, you need to take such changes manually into your custom templates code. But once template is finished we could overwrite old main files with the new ones.

And I´m not sure about when/how to update ManagerManager for new template. Maybe it can be improved to be compatible to old AND new templates by checking for a "Bootstrap"-template first.

When finished and working we can make a poll and ask users / developers to use it as new default template. Me personally I like it!

Just an idea..

**Comment by fourroses666** [comment]
Great solution on hiding the questionmarks pmfx.
To make a little smoother you could use width:0; transition too width:auto;? 
http://front-back.com/css-transition-from-a-fixed-px-width-to-an-auto-width

**Comment by pmfx** [comment]
@Deesen Thanks but working on new "default" theme and keeping track of all the core updates made by devs to the files I have changed is not an option. The only way I see at the moment is to finish new "default" theme and then continue with core updates. Pause updates, finish theme, merge it, test, fix bugs and continue with normal development to the core. Otherwise bugs and conflicts guaranteed... That is why I asked how devs see it and how to accomplish it. I would need a "green light" from devs and someone to keep an eye over it while I develop new theme.

**Comment by yama** [comment]
I thank for your suggestion. Ｉ separate HTML and PHP code before you perform them and make clean code. Because the work takes several months, please wait for a while.

**Comment by Deesen** [comment]
@yama Very exciting! I am looking forward to your ideas. If you need any support (i.e. for "mass-production" of action-files) please let me know.

**Comment by pmfx** [comment]
@yama that is great! Do you want me to send you files that I have changed? Maybe it will be helpful?

**Comment by yama** [comment]
Going to start the current v1.1.1 after release.

@Deesen 
I will refactoring through trial and error . From for a while looking at the situation , please help me .

@pmfx 
Thanks, there is no need for me now . Please help me when it reaches the refactoring.

**Comment by pmfx** [comment]
Closing this issue until we will have template-engine ready.
